Source/Core/Core/Core.cpp

```diff
@@ -244,6 +251,10 @@ bool Init()
 	// Start the emu thread
 	s_emu_thread = std::thread(EmuThread);
 
+#ifdef USE_MEMORYWATCHER
+	s_memory_watcher = new MemoryWatcher;
```

Comment:
No need to add a compile time option, just make this line conditionally will do all of this work.